devdoc/critical_section_requirements.md

```diff
@@ -0,0 +1,47 @@
+`critical_section` requirements
+================
+
+## Overview
+
+`critical_section` is module that provides a few helper functions that let us use a `volatile_atomic int32_t` as a thread synchronization object with `enter_crit_section` and `leave_crit_section` functionality.
+
+It differs from c-pal's `srw_lock` in 2 important respects:
+1. It does not require any `_create` or `_destroy` calls.  It only requires a `volatile_atomic int32_t` which is much simpler to allocate.
+2. All access is exclusive.  There is no concept of shared access.
+
+## Exposed API
+
+```c
+    MOCKABLE_FUNCTION(, void, enter_crit_section, volatile_atomic int32_t*, access_value);

```

Comment:
>enter_crit_section [](http://example.com/codeflow?start=30&length=18)

I think these should be named critical_section_enter and critical_section_leave.

The module name is critical_section

We usually use {module_name}_{verb/action}. Usually ...

/.consider

Reply:
you da boss.